### PR TITLE
Cut v2.40.3

### PR DESCRIPTION
**Pull request by codesome**
Fixes https://github.com/prometheus/prometheus/issues/11585 and cuts a new version at the same time. Do not squash.

Reasoning for the fix: https://github.com/prometheus/prometheus/issues/11585#issuecomment-1324942953

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 2.40.3 / 2022-11-23
+
+* [BUGFIX] TSDB: Fix compaction after a deletion is called. #TODO
```
Comment (roidelapluie) [inline CHANGELOG.md:5]: 11623 :)

**Review by roidelapluie — COMMENTED** [review]
LGTM once changelog is fixed

**Comment by roidelapluie** [comment]
2022-11-23T13:03:20.2156584Z FAIL	github.com/prometheus/prometheus/tsdb	175.796s


is this related?

**Comment by codesome** [comment]
Failed again. Will check on Monday (off for the rest of the week). I saw this error happening on the histogram PR as well, and it magically disappeared later.

**Comment by juliusv** [comment]
:+1:

**Comment by roidelapluie** [comment]
actually it works now

**Comment by codesome** [comment]
Re-run did magic

**Pull request merged**